src/variables.jl

```diff
@@ -161,6 +161,8 @@ end
 Base.iszero(::VariableRef) = false
 Base.copy(v::VariableRef) = VariableRef(v.model, v.index)
 Base.broadcastable(v::VariableRef) = Ref(v)
+Base.getindex(v1::VariableRef, v2::VariableRef) = (v1 == v2 ? one(v1) : zero(v1))
+Base.getindex(v1::VariableRef, v2::VariableRef, v3::VariableRef) = zero(v1)
```

Comment:
These should be `v1 == v2 ? 1.0 : 0.0` and `= 0.0`. (We default to `Float64` here, because it isn't obvious what else to do. We can fix this in future if it becomes a problem.